### PR TITLE
Otp endpoints

### PR DESCRIPTION
**Pull request by spcheema**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,12 +48,55 @@ try {
 }
 ```
 
+**Send OTP**
+
```
Comment (dilantha) [inline README.md:52]: Mention if you receive the OTP id in the response.

File: README.md [hunk 2]
```diff
@@ -48,12 +48,55 @@ try {
...
+try {
+    $response = $otp->send('DESTINATION_NUMBER', '{*code*} is your SMSGlobal verification code.');
+    print_r($response);
+} catch (\Exception $e) {
+    echo $e->getMessage();
+}
+```
+
+**Verify OTP**
+
```
Comment (dilantha) [inline README.md:87]: Explain that you need to send the request id received by the earlier call. Mention what the response might look like and what types of errors to expect.

File: src/Resource/Base.php [hunk 3]
```diff
@@ -20,6 +20,9 @@
 class Base
 {
 
+    // date format accepted by the server
```
Comment (dilantha) [inline src/Resource/Base.php]: How about a doc block here

File: src/Resource/Otp.php [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    /**
+     * @var string
+     */
+    protected $resourceUri = '/otp';
+
+    /**
+     * @param string $to
+     * @param string $text
+     * @param string|null $from
+     * @param int $codeExpirt
```
Comment (dilantha) [inline src/Resource/Otp.php]: `$codeExpiry`

File: src/Resource/Otp.php [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+     * @param \DateTime|null $messageExpiryDateTime
+     *
+     * @return array
+     * @throws AuthenticationException
+     * @throws GuzzleException
+     * @throws InvalidPayloadException
+     * @throws InvalidResponseException
+     * @throws ResourceNotFoundException
+     */
+    public function send(string $to, string $text, string $from = null, string $codeExpiry = null, string $length = null, \DateTime $messageExpiryDateTime = null): array
```
Comment (dilantha) [inline src/Resource/Otp.php:37]: Looks like a lot of parameters to get wrong. How about using a `sendOptions` or `sendRequest` class for these?

File: src/Resource/Otp.php [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+    {
+        $origin = !empty($from) ? $from : '';
+        $codeExpiry = !empty($codeExpiry) ? $codeExpiry : '';
+        $length = !empty($length) ? $length : '';
+        $messageExpiryDateTime = !empty($messageExpiryDateTime) ? $messageExpiryDateTime->format(self::DATE_FORMAT) : '';
```
Comment (dilantha) [inline src/Resource/Otp.php:42]: These validations seem quite basic. Can we use the Symfony validator to do a better job?

File: src/Resource/Otp.php [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+     */
+    public function verfiyById(string $requestId, string $code)
+    {
+        $origin = !empty($from) ? $from : '';
+
+        $uri = $this->prepareApiUri($this->resourceUri . '/' . $requestId);
+
+        $jsonPayload = json_encode([
+            'code' => $code,
+        ], JSON_FORCE_OBJECT);
```
Comment (dilantha) [inline src/Resource/Otp.php]: $jsonPayload = json_encode(compact('code'), JSON_FORCE_OBJECT);

**Review by dilantha — COMMENTED** [review]
Looks good with a few suggestions

File: src/Resource/Otp.php (hunk 6 above)
Comment (spcheema) [reply src/Resource/Otp.php:42]: we already got fool proof validation on the api side so I dont feel any need of this.

File: src/Resource/Otp.php (hunk 5 above)
Comment (spcheema) [reply src/Resource/Otp.php:37]: Existing send sms method is the same fashion so I dont want to introduce the new way in the minor version. May be next major release we change both.

File: src/Resource/Otp.php (hunk 7 above)
Comment (spcheema) [reply src/Resource/Otp.php]: I like it

**Pull request merged**